src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -4009,4 +4009,17 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
 
         fn foo() -> impl FooBar {}
     """)
+
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test E0116 inherent impls should be in same crate`() = checkByFileTree("""
+    //- lib.rs
+        pub struct ForeignStruct {}
+    //- main.rs
+        /*caret*/
+        use test_package::ForeignStruct;
+        impl <error descr="Cannot define inherent `impl` for a type outside of the crate where the type is defined [E0116]">ForeignStruct</error> {}
```

Comment:
What about error annotations for aliased types? E.g.
```Rust
type Foo = ForeignStruct;
impl Foo {}
```

Reply:
Thanks, updated tests